### PR TITLE
core/btree: improve documentation

### PR DESCRIPTION
**Pull request by jussisaurio**
This PR should have no functional changes, just variable renaming and comments

Using `///` comment format for better IDE support

### CODE REVIEW

File: core/storage/btree.rs [hunk 1]
```diff
@@ -734,41 +763,51 @@ impl BTreeCursor {
...
-            page.write_u16(BTREE_HEADER_OFFSET_FREEBLOCK, offset);
+            page.write_u16(offset as usize, first_block); // next freeblock = previous first freeblock
+            page.write_u16(offset as usize + 2, len); // size of this freeblock
+            page.write_u16(PAGE_HEADER_OFFSET_FREEBLOCK, offset); // first freeblock in page = this block
             return;
         }
 
+        // if we clear space that is at the start of the cell content area,
+        // we need to update the cell content area pointer forward to account for the removed space
+        // FIXME: is offset ever < cell_content_area?
```
Comment (jussisaurio) [inline core/storage/btree.rs:798]: ?

File: core/storage/btree.rs [hunk 2]
```diff
@@ -734,41 +763,51 @@ impl BTreeCursor {
...
         if offset <= page.cell_content_area() {
-            // extend boundary of content area
-            page.write_u16(BTREE_HEADER_OFFSET_FREEBLOCK, page.first_freeblock());
-            page.write_u16(BTREE_HEADER_OFFSET_CELL_CONTENT, offset + len);
+            page.write_u16(PAGE_HEADER_OFFSET_FREEBLOCK, page.first_freeblock()); // why is this here?
```
Comment (jussisaurio) [inline core/storage/btree.rs:800]: ?

File: core/storage/btree.rs [hunk 3]
```diff
@@ -1423,25 +1479,26 @@ impl BTreeCursor {
...
             cell_payload.extend_from_slice(record_buf.as_slice());
             cell_payload.resize(cell_payload.len() + 4, 0);
             return;
         }
         log::debug!("fill_cell_payload(overflow)");
 
-        let min_local = self.min_local(page_type);
-        let mut space_left = min_local + (record_buf.len() - min_local) % (self.usable_space() - 4);
+        let payload_overflow_threshold_min = self.payload_overflow_threshold_min(page_type);
+        let mut space_left = payload_overflow_threshold_min
```
Comment (jussisaurio) [inline core/storage/btree.rs:1497]: I don't understand this calculation and it should be commented

File: core/storage/sqlite3_ondisk.rs [hunk 4]
```diff
@@ -1170,28 +1271,45 @@ pub fn begin_write_wal_header(io: &Rc<dyn File>, header: &WalHeader) -> Result<(
...
 ) -> (bool, usize) {
-    if payload_size <= max_local {
+    if payload_size <= payload_overflow_threshold_max {
         return (false, 0);
     }
 
-    let mut space_left = min_local + (payload_size - min_local) % (usable_size - 4);
-    if space_left > max_local {
-        space_left = min_local;
+    let mut space_left = payload_overflow_threshold_min
```
Comment (jussisaurio) [inline core/storage/sqlite3_ondisk.rs:1287]: I don't understand this calculation and it should be commented

File: core/storage/btree.rs [hunk 5]
```diff
@@ -67,11 +74,16 @@ enum WriteState {
...
+    /// State of the write operation state machine.
     state: WriteState,
+    /// Pages allocated during the write operation due to balancing.
     new_pages: RefCell<Vec<PageRef>>,
+    /// Scratch space used during balancing.
     scratch_cells: RefCell<Vec<&'static [u8]>>,
+    /// Bookkeeping of the rightmost pointer so the PAGE_HEADER_OFFSET_RIGHTMOST_PTR can be updated.
     rightmost_pointer: RefCell<Option<u32>>,
-    page_copy: RefCell<Option<PageContent>>, // this holds the copy a of a page needed for buffer references
+    /// Copy of the current page needed for buffer references.
```
Comment (jussisaurio) [inline core/storage/btree.rs:85]: I don't really know what this means tbh

Comment (pereman2) [reply core/storage/btree.rs:85]: What do you mean?

File: core/storage/btree.rs (hunk 3 above)
Comment (pereman2) [reply core/storage/btree.rs:1497]: I don't remember right now. Might need to look at it again. There are things that are basically 1:1 copies of what sqlite does that's why names are like that

File: core/storage/btree.rs (hunk 2 above)
Comment (pereman2) [reply core/storage/btree.rs:800]: Looks like a noop

File: core/storage/btree.rs (hunk 1 above)
Comment (pereman2) [reply core/storage/btree.rs:798]: Mental notes while developing. I don't remember what idea I had in mind xd

File: core/storage/btree.rs (hunk 5 above)
Comment (jussisaurio) [reply core/storage/btree.rs:85]: I guess I meant the comment would help more if there was a description of why a copy is needed in the context of balancing, but it's not a big deal

File: core/storage/btree.rs (hunk 3 above)
Comment (jussisaurio) [reply core/storage/btree.rs:1497]: I can try to find the places in sqlite code that these are direct c-to-rust translations of and attach them as comments

Comment (jussisaurio) [reply core/storage/btree.rs:1497]: also we can keep calling these `max_local` and `min_local` if it helps in referencing back to sqlite later. i just found those names confusing and tried to come up w something else, although it's difficult to translate these into concise variable names:

```
max_local = maximum payload size before the content must spill to overflow pages
min_local = minimum payload size when the content is ALLOWED to spill to overflow pages
```

File: core/storage/btree.rs (hunk 1 above)
Comment (jussisaurio) [reply core/storage/btree.rs]: rewrote the FIXME to make it clearer what I mean. I don't want to change runtime code in this PR.

File: core/storage/btree.rs (hunk 2 above)
Comment (jussisaurio) [reply core/storage/btree.rs]: Added a better FIXME for this. I don't want to change runtime code in this PR

File: core/storage/btree.rs (hunk 3 above)
Comment (jussisaurio) [reply core/storage/btree.rs:1501]: Added a reference to sqlite source in https://github.com/tursodatabase/limbo/pull/539/commits/c417fe788083f2dfa20970c9954facaabd9cf654

File: core/storage/sqlite3_ondisk.rs (hunk 4 above)
Comment (jussisaurio) [reply core/storage/sqlite3_ondisk.rs:1290]: Added a reference to sqlite source in https://github.com/tursodatabase/limbo/pull/539/commits/c417fe788083f2dfa20970c9954facaabd9cf654

**Review by pereman2 — APPROVED** [review]
Looks good, can you run `cd test; cargo test -- --include-ignored --nocapture` to ensure everything is ok?

**Pull request closed without merging**